gitbutler-ui/src/lib/components/CommitDialog.svelte

```diff
@@ -39,31 +40,55 @@
 	export let cloud: ReturnType<typeof getCloudApiClient>;
 	export let user: User | undefined;
 	export let selectedOwnership: Writable<Ownership>;
+	export const expanded = persisted<boolean>(false, 'commitBoxExpanded_' + branch.id);
 
 	const aiGenEnabled = projectAiGenEnabled(projectId);
 	const runCommitHooks = projectRunCommitHooks(projectId);
 	const currentCommitMessage = projectCurrentCommitMessage(projectId, branch.id);
-	export const expanded = persisted<boolean>(false, 'commitBoxExpanded_' + branch.id);
 
-	let commitMessage: string = get(currentCommitMessage) || '';
-	let isCommitting = false;
-	let textareaElement: HTMLTextAreaElement;
-	$: if (textareaElement && commitMessage && expanded) {
-		textareaElement.style.height = 'auto';
-		textareaElement.style.height = `${textareaElement.scrollHeight + 2}px`;
+	function getCommitMessageTitleAndDescription(commitMessage: string) {
+		// Split the commit message into title and description
+		// get the first line as title and the rest as description
+		const [summary, ...description] = commitMessage.split('\n');
+		return {
+			summary: summary || '',
+			// remove top new line
+			description: description.join('\n').replace(/^\n/, '') || ''
```

Comment:
Perhaps we could `.trim()` rather than `.replace(/^\n/,'')` since it would remove both leading and trailing whitespace?

Reply:
Make sense

Reply:
Done, description returning as an array, so I just take the first item `description[0]`